catalog/CMakeLists.txt

```diff
@@ -36,6 +36,7 @@ add_library(quickstep_catalog_CatalogRelationSchema
             CatalogRelationSchema.hpp)
 add_library(quickstep_catalog_CatalogTypedefs ../empty_src.cpp CatalogTypedefs.hpp)
 add_library(quickstep_catalog_PartitionScheme PartitionScheme.cpp PartitionScheme.hpp)
+add_library(quickstep_catalog_IndexScheme IndexScheme.cpp IndexScheme.hpp)
```

Comment:
Please move this line between `quickstep_catalog_CatalogTypedefs` and `quickstep_catalog_PartitionScheme`, as we sort them in alphabet order.

Note that we should also reorder `quickstep_catalog_NUMAPlacementScheme` to the right place in another PR.